### ISSUE TITLE
Add a way to sort projects in the dashboard

### ISSUE THREAD

**Issue by almet** [spiral-project/ihatemoney#112]
Currently, when you access /dashboard you have some stats but that's not possible to sort them so they're pretty useless.

**Comment by almet** [comment]
I'm mainly thinking we should redo the project's dashboard, in order to display some stats, to ease the deletion of old projects (or sending an email), etc.

I believe this is a good candidate for a first bug :-)

**Comment by bmatticus** [comment]
I'm taking a stab at this one, just took the dashboard and implemented DataTables as a quick/easy way to add sorting/searching/pagination to the dashboard view.  Created a pull request with updates referencing this issue.

**Comment by bmatticus** [comment]
Bringing this back here for now. I, reading back over this, think there were ideas of some more additions as well? 

1. Add sorting; pagination and searching comes "free" if DataTables is implemented and we delegate this work to the browser.
2. Make project name a link into the project to quickly drive into it to perform other tasks?
3. Additions to actions possibly? I played with adding invite here, but the paper plane svg is white and does not work well in this table as is. I can add backgrounds/etc but won't match the rest of the theme.

I'm willing to look further into a back-end driven method for sorting if it is more desirable. Using JS just makes things a bit simpler overall.

![image](https://user-images.githubusercontent.com/36943357/75264207-17ec0280-57bd-11ea-89ed-8a7948cf7711.png)

**Comment by almet** [comment]
Nice :-)

>  Add sorting; pagination and searching comes "free" if DataTables is implemented and we delegate this work to the browser.

I think it's good, yeah!

> Make project name a link into the project to quickly drive into it to perform other tasks?

Why not :-)

> Additions to actions possibly? I played with adding invite here, but the paper plane svg is white and does not work well in this table as is. I can add backgrounds/etc but won't match the rest of the theme.

I'm not sure it would be the right place for this, we probably should let the actions to the project itself, don't you think?

**Comment by bmatticus** [comment]
> I'm not sure it would be the right place for this, we probably should let the actions to the project itself, don't you think?

Agree, there was a mention of making it easier to send emails from the dashboard possibly in the second comment? 

> I'm mainly thinking we should redo the project's dashboard, in order to display some stats, to ease the deletion of old projects (or sending an email), etc.

Maybe a difference email, or maybe driving into the project from here more easily is the proper answer?

**Comment by bmatticus** [comment]
I've found SQLAlchemy-DataTables that allows the use of SQLAlchemy with datatables to drive backend queries and offload some of the complexities of the queries. I have tested this with the dashboard and it works, but the formatting of the table has to be done in a slightly different, less than ideal, manner. 

This can be applied to the existing bills page too, potentially. I have noticed that now that it does queries in the backend via ajax to update the table performance can be kind of spotty. Though my environment is running a quick mysql docker and is running on an older NUC. I feel like most of these datasets will have a relatively low memory footprint and may be fine to load to the client to handle.

I have also tested both approaches side by side using mobile, on an iPhone XS and 2019 iPad at least, and both seem viable for desktop and mobile.
